src/lib/flexbox/api/layout-gap.ts

```diff
@@ -71,11 +90,30 @@ export class LayoutGapDirective extends BaseFxDirective implements AfterContentI
     this._updateWithValue();
   }
 
+  ngOnDestroy() {
+     super.ngOnDestroy();
+     if (this._layoutWatcher) {
+       this._layoutWatcher.unsubscribe();
+     }
+   }
+
   // *********************************************
   // Protected methods
   // *********************************************
 
   /**
+   * Cache the parent container 'flex-direction' and update the 'flex' styles
+   */
+  private _onLayoutChange(direction) {
+    this._layout = (direction || '').toLowerCase().replace('-reverse', '');
```

Comment:
Would a reverse row layout use `margin-right` instead of `margin-left`?  This information might be useful for RTL row layouts.